playersettings-plugin/src/main/java/me/limbo56/playersettings/settings/defined/FlySetting.java

```diff
@@ -35,15 +35,24 @@ public int getSlot() {
     }
 
     @Override
-    public boolean getDefaultValue() {
-        return false;
+    public int getDefaultValue() {
+        return 0;
+    }
+
+    @Override
+    public int getMaxValue() {
+        return 10;
     }
 
     public static class FlySettingCallback implements SettingCallback {
         @Override
-        public void notifyChange(Setting setting, Player player, boolean newValue) {
-            player.setAllowFlight(newValue);
-            player.setFlying(newValue);
+        public void notifyChange(Setting setting, Player player, int newValue) {
+            GameMode gm = player.getGameMode();
+            if (gm != GameMode.CREATIVE && gm != GameMode.SPECTATOR) {
```

Comment:
We don't really mean to remove creative players' ability to fly, do we?
This makes it act as a speed multiplier only when creative